libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2213,7 +2213,7 @@ void TypeChecker::typeCheckABIEncodeCallFunction(FunctionCall const& _functionCa
 				externalFunctionType->declaration().scope() == m_currentContract
 			)
 				msg += " Did you forget to prefix \"this.\"?";
```

Comment:
From the implementation, looks like there should be more affected cases and indeed I found some. Could you add these as tests?

```solidity
function g(uint) {}

function f() {
    abi.encodeCall(g, (1));
}
```

```solidity
library L {
    function f() external {}
}

function f() {
    abi.encodeCall(L.f, (1));
}
```

Reply:
Confirmed that these cases are also affected. Added tests.